### ISSUE TITLE
minetest.serialize() is not handling large numeric indexes right leading to forceloading additional blocks

### ISSUE THREAD

**Issue by t4im** [minetest/minetest#2206]
`minetest.serialize()` appears to use `tostring()` on the keys, thereby larger numeric indexes are prone to be rendered in scientific notation when reaching a certain size.

`minetest.hash_node_positon()` supplies a 48bit number which is large enough to trigger that behavior of tostring()

Not just does this mean a (ever so slightly) increase in collision chances of hashmaps created this way; it will render the deserialized key useless from being used with  `minetest.get_position_from_hash()`

TLDR; `minetest.deserialize(minetest.serialize(data)) ~= data` for its elements with numeric indexes of a certain size

**Comment by ShadowNinja** [comment]
`string.format("%d", num)` seems to fix this.

**Comment by t4im** [comment]
Can confirm that fixing it.

Currently forceloading is persistently keeping blocks loaded that it didn't intent to due to `minetest.get_position_from_hash()` loading a neighbouring block every now and then.
So this might need some additional handling in forceloading.lua

**Comment by ShadowNinja** [comment]
47aca6f6d164347de950a46b11e1774445cdb9e3
